float32[6] x
float32[6] y
float32[6] th
bool[6] found
float32 ball_x
float32 ball_y
